packages/components/src/ComboBox/src/ComboBox.tsx

```diff
@@ -0,0 +1,337 @@
+        popoverProps,
+        prefix,
+        size: sizeProp,
+        style: styleProp,
+        triggerProps,
+        ...otherProps
+    } = ownProps;
+    const inputRef = useObjectRef(mergeRefs(userProvidedInputRef, props.inputRef !== undefined ? props.inputRef : null));
+    const inputContext = useSlottedContext(InputContext);
+    // Make sure to merge the input ref with the context ref from the InputContext.
+    const mergedInputRefs = inputContext?.ref ? mergeRefs(inputRef, inputContext?.ref) : inputRef;
+    const triggerRef = useRef<HTMLDivElement>(null);
+    const buttonRef = useRef<HTMLButtonElement>(null);
+
+    // Make menu width match trigger width.
+    const [triggerWidth, setTriggerWidth] = useState<string | null>(null);
+    const onResize = useCallback(() => {
+        if (triggerRef.current) {
+            setTriggerWidth(triggerRef.current.offsetWidth + "px");
+        }
+    }, [triggerRef]);
+
+    useResizeObserver({
+        ref: triggerRef,
+        onResize: onResize
+    });
+    
+    const { stylingProps: triggerStylingProps, ...triggerOwnProps } = useStyledSystem(triggerProps ?? {});
+    const { 
+        className: triggerClassName,
+        style: triggerStyleProp,
+        ...otherTriggerProps
+    } = triggerOwnProps;
+    const { stylingProps: popoverStylingProps, ...popoverOwnProps } = useStyledSystem(popoverProps ?? {});
+    const { 
+        placement = "bottom start",
+        style: popoverStyleProp,
+        ...otherPopoverProps 
+    } = popoverOwnProps;
+    
+    const size = useResponsiveValue(sizeProp) ?? "sm";
+    const isFluid = useResponsiveValue(isFluidProp) ?? false;
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalComboBoxCssSelector,
+        cssModule(
+            styles,
+            "hop-ComboBox",
+            isFluid && "fluid",
+            size
+        ),
+        stylingProps.className
+    );
+
+    const triggerClassNames = composeClassnameRenderProps(
+        triggerClassName,
+        cssModule(
+            styles,
+            "hop-ComboBox__trigger",
+            size
+        ),
+        triggerStylingProps.className
+    );
+
+    const buttonClassNames = composeClassnameRenderProps(
+        cssModule(
+            styles,
+            "hop-ComboBox__button"
+        )
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const triggerStyle = composeRenderProps(triggerStyleProp, prev => {
+        return {
+            ...triggerStylingProps.style,
+            ...prev
+        };
+    });
+
+    const popoverStyle = composeRenderProps(popoverStyleProp, prev => {
+        return {
+            ...popoverStylingProps.style,
+            ...prev,
+            "--custom-trigger-width": triggerWidth
+        };
+    });
+    
+    const handleMouseDown: MouseEventHandler<HTMLElement> = useCallback(e => {
```

Comment:
```suggestion
    const handleMouseDown: MouseEventHandler<HTMLDivElement> = useCallback(e => {
```

Not sure about this one, but triggerRef is a div, so we can specify it.